enhancements/management-workload-partitioning.md

```diff
@@ -0,0 +1,549 @@
+---
+title: management-workload-partitioning
+authors:
+  - "@dhellmann"
+  - "@mrunalp"
+  - "@browsell"
+  - "@haircommander"
+  - "@rphillips"
+reviewers:
+  - "@deads2k"
+  - "@staebler"
+  - TBD
+approvers:
+  - "@smarterclayton"
+  - "@derekwaynecarr"
+  - "@markmc"
+creation-date: 2021-03-18
+last-updated: 2021-03-18
+status: implementable
+see-also:
+  - "/enhancements/single-node-production-deployment-approach.md"
+replaces:
+  - https://github.com/openshift/enhancements/pull/628
+---
+
+# Management Workload Partitioning
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+This enhancement describes an approach to allow us to isolate the
+control plane services to run on a restricted set of CPUs. This will
+be especially useful for resource-constrained enviornments, such as
+single-node production deployments, where the user wants to reserve
+most of the CPU resources for their own workloads and needs to
+configure OpenShift to run on a fixed number of CPUs within the host.
+
+One example of this use case is seen in telecommunication service
+providers implementation of a Radio Access Network (RAN). This use case
+is discussed in more detail below.
+
+## Motivation
+
+In constrained environments, management workloads, including the
+OpenShift control plane, need to be configured to use fewer resources
+than they might by default in normal clusters. After examining
+[various approaches for scaling the resource
+requests](https://github.com/openshift/enhancements/pull/628), we are
+reframing the problem to allow us to solve it a different way.
+
+Customers who want us to reduce the resource consumption of management
+workloads have a fixed budget of CPU cores in mind. We want to use
+normal scheduling capabilities of kubernetes to manage the number of
+pods that can be placed onto those cores, and we want to avoid mixing
+management and normal workloads there.
+
+### Goals
+
+* This enhancement describes an approach for configuring OpenShift
+  clusters to run with management workloads on a restricted set of
+  CPUs.
+* Clusters built in this way should pass the same Kubernetes and
+  OpenShift conformance and functional end-to-end tests as single-node
+  deployments that are not isolating the management workloads.
+* We have a goal of running on 4 hyperthreads today, but we do not
+  know what constraints we might be given in the future so we need a
+  solution that is not tied to the current known limit.
+* We want a general approach, that can be applied to all OpenShift
+  control plane components.
```

Comment:
control plane AND per-node components

Reply:
when running SNO, management components implicitly include per node and control components.
when running compact clusters (schedulable control hosts), it likely means the same.
when running non-compact, management in this context means per node only (and not control).

Reply:
Is it safe to assume that in a non-compact cluster the user is going to want to dedicate an entire host (or 3) to the management workloads?

Reply:
maybe.  maybe not.  i think it’s orthogonal and if you create a management cpu pool you get all management workloads in there.  if we want to get clever later we come back with a more clever proposal